WcaOnRails/app/views/competitions/_index_table.html.erb

```diff
@@ -17,7 +17,7 @@
         <% else %>
           <%= icon('hourglass-start', title: "Starts in #{ pluralize((competition.start_date - Date.today).to_i, "day") }", data: { toggle: "tooltip" }) %>
         <% end %>
-        <%= wca_date_range(competition.start_date, competition.end_date)  %>
+        <%= competition.start_date && competition.end_date ? wca_date_range(competition.start_date, competition.end_date) : "Unscheduled" %>
```

Comment:
I've created https://github.com/cubing/worldcubeassociation.org/issues/604 to track the additional Competition validations. @pingskills, please don't worry about taking care of that in this PR.